Fix TypeError by checking if duration is not Finite before setting cu…

spran180 opened the pull request:
…rrentTime.

<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Fixes #4156 by @sentry-io

## Description

This pull request resolves the TypeError that occurs when the 'currentTime' property is applied to 'HTMLMediaElement' with a non-finite value. This issue arises when the 'duration' value is 'NaN'.

## Changes
- Added checks to ensure that `duration.value` is a finite number before setting `localAudio.currentTime`.
- If `duration.value` is not a finite number, it checks if `localAudio.duration` is a finite number and uses that instead.
- It logs the warning if neither `duration.value` nor `localAudio.duration` is finite number, and skips the seek operation.

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [x] TypeError is handleSeeked while checking the duration value is finite.
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices] (https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines).
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`./ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`./ov just catalog/generate-docs media-props`
      for the catalog or `./ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat commented on frontend/src/components/VAudioTrack/VAudioTrack.vue:
```diff
@@ -450,7 +450,15 @@ export default defineComponent({
+        if(Number.isFinite(duration.value)){
+          localAudio.currentTime = frac * duration.value
+        }
+        else if(Number.isFinite(localAudio.duration)){
```
```suggestion
        else if (Number.isFinite(localAudio.duration)){
```

obulat commented on frontend/src/components/VAudioTrack/VAudioTrack.vue:
```diff
@@ -450,7 +450,15 @@ export default defineComponent({
        */
       if (localAudio) {
-        localAudio.currentTime = frac * duration.value
+        if(Number.isFinite(duration.value)){
```
```suggestion
        if (Number.isFinite(duration.value)){
```

obulat commented on frontend/src/components/VAudioTrack/VAudioTrack.vue:
```diff
@@ -450,7 +450,15 @@ export default defineComponent({
+        else if(Number.isFinite(localAudio.duration)){
+          localAudio.currentTime = frac * localAudio.duration
+        }
+        else{
```
```suggestion
        else {
```

obulat reviewed (changes requested):
Thank you for your PR, @spran180 !
There are some linting problems that prevent the rest of the CI from running. Could you run `just lint` command in the top-level folder (or `./ov just lint`, if you used ov script)?
I suspect that the lack of spaces before the brackets are the problems, but there could be something else, too

sarayourfriend commented on frontend/src/components/VAudioTrack/VAudioTrack.vue:
```diff
@@ -450,7 +450,15 @@ export default defineComponent({
+          localAudio.currentTime = frac * localAudio.duration
+        }
+        else {
+          console.warn('Duration is not finite. Skipping seek.')
```
@obulat should this use `warn` imported from `~/utils/logger`?

spran180 replied:
Hi @sarayourfriend,

I noticed that the use `warn` isn't imported from `~/utils/logger`. Should I import it, or do you want me to remove the `warn` usage?

Thanks!

sarayourfriend replied:
Oops, I mean `~/utils/console`, not `logger`, apologies.

But. it would need to be imported, yes. I just wanted to check with Olga whether it should be used here rather than `console.warn` directly. I don't know whether we specifically want to always log this on the client for some reason, for example, which `~/utils/console` would suppress.

zackkrida replied:
@obulat can you review this discussion and make an issue if necessary? Thanks!

spran180 commented:
Hi @obulat , I ran the pre-commit checks instead of the `just lint` command since I don't have Docker permissions. 

![Screenshot 2024-07-01 223215-copy](https://github.com/WordPress/openverse/assets/83678994/eca8e452-602f-4a03-a175-154dd8314b13)

Please review and let me know if further changes are needed. Thanks!

sarayourfriend commented:
It's curious @spran180, looks like pre-commit skipped the prettier hook, which is the one that fails in CI.

Can you try `pre-commit run --all-files`, if you ran it differently before?

spran180 commented:
Hi @sarayourfriend,

I've addressed the issues you pointed out. I ran `prettier` on the `frontend/src/components/VAudioTrack/VAudioTrack.vue` file and committed the changes. Additionally, I made a few more formatting adjustments where necessary. Everything should be in order now. Please let me know if there are any further changes or checks required.

sarayourfriend reviewed (approved):
LGTM! I suppose it's rather hard to add a test to reproduce this issue, so I'm not too worried about it. Thanks for sorting out the linting @spran180 and for the contribution generally :pray:

zackkrida reviewed (approved):
Thanks, @spran180!
